test/util/scale/utils.spec.ts

```diff
@@ -0,0 +1,74 @@
+import { describe, expect } from 'vitest';
+import { compose, curry, map, memoize, range, reverse } from '../../../src/util/scale/util/utils';
+
+describe('functional utilities', () => {
+  describe('curry', () => {
```

Comment:
💭 Love the tests, I am a bit sceptical as to whether we need these helper functions, and whether we could just get by with a simpler solution that just might be more verbose - but easier to read and maintain.

Reply:
These were all brought over from recharts-scale

Your concern might be true (we don't need these), but we should get more test coverage first to ensure nothing breaks